web/src/main/java/org/cbioportal/web/ClinicalAttributeController.java

```diff
@@ -114,4 +121,39 @@
         return new ResponseEntity<>(clinicalAttributeService.getClinicalAttribute(studyId, clinicalAttributeId),
                 HttpStatus.OK);
     }
+    
+    @RequestMapping(value = "/clinical-attributes/fetch", method = RequestMethod.POST,
+                    produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get all clinical attributes in specified sampleIdentifiers or sampleListID with clinical attribute count")
+    public ResponseEntity<List<ClinicalAttribute>> getAllClinicalAttributesInStudies(
+    			 @ApiParam(required = false, value = "List of sample identifiers")
+             @RequestBody List<SampleIdentifier> sampleIdentifiers,
+             @ApiParam("Sample list ID e.g. acc_tcga_all")
+             @RequestParam(required = false) String sampleListId,
```

Comment:
As I said earlier, you need to create a filter like https://github.com/cBioPortal/cbioportal/blob/rc/web/src/main/java/org/cbioportal/web/parameter/DiscreteCopyNumberFilter.java and put sampleIdentifiers, and sampleListId there, and make sure only one of them is filled using `@AssertTrue`. And also check `@Size(min, max)` just like in the example.